### ISSUE TITLE
Unexpected change to metrics sent to backends with new dbengine

### ISSUE THREAD

**Issue by cakrit** [netdata/netdata#6039]
Could it be that database change changed behavior to backends? I use graphite and average the netdata data, which it will send every 60 seconds. When my netdata automatically updated via cron on the 16th at 6.25am, my graphs started to change as seen in the picture. In the meantime I played and looked around to finally reinstall netdata from commit 0d5fa83e224afa6d530ed6a5e6e6fe35a8af955e (fixing segmentation fault). The graphs looks back to normal (to me at least) again.
![load](https://user-images.githubusercontent.com/2340029/57907008-462ad200-787c-11e9-9f1b-9be0721eda57.png)

_Originally posted by @idef1x in https://github.com/netdata/netdata/pull/5282#issuecomment-493335227_

**Comment by cakrit** [comment]
> Did you explicitly set memory mode = dbengine to change the database engine? The new database is not on by default.

> No...it's on default setting (# memory mode = save) and it happened right after the netdata-updater cron job. So looking through the commits between the previous nightly update (the 15th) and the one on the 16th, I thought it might something to do with the database change..(the others didn't seem relevant to me, but I am not a programmer)..

**Comment by cakrit** [comment]
Please put here the configuration for your backend, excluding the IP of course.

**Comment by idef1x** [comment]
Here it is:

> [backend]                                                                                                                                                                       
        # host tags = 
        enabled = yes                                                                                                                                                           
        data source = average                                                                                                                                                   
        type = graphite                                                                                                                                                         
        destination = <IP of graphite server>                                                                                                                                             
        prefix = netdata                                                                                                                                                        
        hostname = panoramix                                                                                                                                                    
        update every = 60                                                                                                                                                       
        buffer on failures = 5                                                                                                                                                  
        timeout ms = 20000                                                                                                                                                      
        send names instead of ids = yes                                                                                                                                         
        #send charts matching = system.* disk.* zfs.* cgroup_lxc.payload_*
        send charts matching = *                                                                                                                                                
        # send hosts matching = localhost *

**Comment by mfundul** [comment]
@idef1x can you please update netdata to the latest code and verify that your issue is resolved?

**Comment by cakrit** [comment]
We won't close it until we get verification that it works.

**Comment by cakrit** [comment]
This one may be fixed, but we have another issue. We're not sending all the metrics. I also checkout out commit 0d5fa83 and immediately got the charts that I was missing with the latest code. So we need another PR as well.

**Comment by idef1x** [comment]
Just reinstalled it from scratch (except config file) and data is coming in again...lets wait a while to see the final results. 
I'll look for other charts that might be missing as well then, although the ones I defined in grafana at the moment seems to be there..

**Comment by cakrit** [comment]
Seems like I am getting other metrics as well now, I'll do a new installation of graphite and try again with the new code, just to be sure...

**Comment by idef1x** [comment]
So far it looks good to me, besides my lxd containers io doesn't show anything (lxc.payload container is where I am looking), but these are flat liners on the netdata dashboard as well, so that must be something else

**Comment by cakrit** [comment]
We verified that the issue with graphite not accepting all the metrics we send is not related to the new netdata version. After several tests, we saw that even with that older commit, netdata can still be sending metrics to graphite and graphite doesn't add them. So there may be something going on with the latest version of graphite we used in the tests and we may need to change something in our backend code. In any case, I will open a new issue for this investigation.
